src/daemon/ca/manager.rs

```diff
@@ -1334,48 +1374,111 @@ impl CaManager {
-                .into_reply()
-                .map_err(Error::custom)
+            let user_agent = Some("local-child".to_string());
+
+            self.rfc6492_process_request(&parent, message, user_agent, &self.system_actor)
+                .await
         } else {
-            let response = self
-                .send_protocol_msg_and_validate(
-                    signing_key,
-                    parent_res.service_uri(),
-                    parent_res.id_cert(),
-                    rfc6492::CONTENT_TYPE,
-                    msg.into_bytes(),
-                    cms_logger,
-                )
+            // Set up a logger for CMS exchanges. Note that this logger is always set
+            // up and used, but.. it will only actually save files in case the given
+            // rfc6492_log_dir is Some.
+            let cms_logger = CmsLogger::for_rfc6492_sent(
+                self.config.rfc6492_log_dir.as_ref(),
+                message.sender(),
+                message.recipient(),
+            );
+
+            let cms = self.signer.create_rfc6492_cms(message, signing_key)?.to_bytes();
+
+            let res_bytes = self
+                .post_protocol_cms_binary(&cms, service_uri, provisioning::CONTENT_TYPE, &cms_logger)
                 .await?;
 
-            rfc6492::Message::from_signed_message(&response)
-                .map_err(Error::custom)?
-                .into_reply()
-                .map_err(Error::custom)
+            match ProvisioningCms::decode(&res_bytes) {
+                Err(e) => {
+                    cms_logger.err(format!("Could not decode CMS: {}", e))?;
+                    Err(Error::Rfc6492(e))
+                }
+                Ok(cms) => match cms.validate(parent_res.id_cert()) {
+                    Err(e) => {
+                        cms_logger.err(format!("Response invalid: {}", e))?;
+                        Err(Error::Rfc6492(e))
+                    }
+                    Ok(()) => Ok(cms.into_message()),
+                },
+            }
+        }
+    }
+
+    async fn post_protocol_cms_binary(
+        &self,
+        msg: &Bytes,
+        service_uri: &ServiceUri,
+        content_type: &str,
+        cms_logger: &CmsLogger,
+    ) -> KrillResult<Bytes> {
+        cms_logger.sent(msg)?;
+
+        let timeout = self.config.post_protocol_msg_timeout_seconds;
+
+        match httpclient::post_binary_with_full_ua(service_uri.as_str(), msg, content_type, timeout).await {
+            Err(e) => {
+                cms_logger.err(format!("Error posting CMS: {}", e))?;
+                Err(Error::HttpClientError(e))
+            }
+            Ok(bytes) => {
+                cms_logger.reply(&bytes)?;
+                Ok(bytes)
+            }
+        }
+    }
+
+    /// Returns the handle of the local parent for this specific ServiceUri, and the
+    /// configured base (service) uri. Provided that this indeed maps back to this
+    /// same server and it is an rfc6492 style krill uri.
```

Comment:
s/uri/URI/, and s/krill/Krill/. Also, elsewhere you have corrected rfcNNNN occurrences to RFC NNNN.